app/presenters/sufia/file_set_presenter.rb

```diff
@@ -6,6 +6,14 @@ class FileSetPresenter < ::CurationConcerns::FileSetPresenter
     delegate :depositor, :keyword, :date_created, :date_modified, :itemtype,
              to: :solr_document
 
+    def page_title
+      title.first
+    end
+
+    def link_name
+      current_ability.can?(:read, id) ? page_title : 'File'
+    end
+
```

Comment:
OK, I see where the confusion arose. I was saying use a view test that's "integration-ish" because feature tests are slow. Looks like that's already been done, so I'm happy. :) But, to @mjgiarlo's point, since the feature test is already there, adding an additional like to check for the link is probably the better option. However, it's fine the way it is!